packages/react-native-reanimated/Common/cpp/Fabric/ShadowTreeCloner.cpp

```diff
@@ -6,60 +6,67 @@
 
 namespace reanimated {
 
-ShadowNode::Unshared cloneShadowTreeWithNewProps(
-    const ShadowNode::Shared &oldRootNode,
-    const ShadowNodeFamily &family,
-    RawProps &&rawProps) {
-  // adapted from ShadowNode::cloneTree
+ShadowNode::Unshared cloneShadowTreeWithNewPropsRecursive(
+    std::unordered_map<const ShadowNodeFamily *, std::vector<int>> &childrenMap,
+    const ShadowNode::Shared &shadowNode,
+    std::unordered_map<
+        const ShadowNodeFamily *,
+        std::vector<std::shared_ptr<RawProps>>> &propsMap) {
+  auto family = &shadowNode->getFamily();
+  auto children = shadowNode->getChildren();
+  auto &affectedChildren = childrenMap[family];
```

Comment:
Let's use `const` wherever possible.
```suggestion
  const auto family = &shadowNode->getFamily();
  const auto children = shadowNode->getChildren();
  const auto &affectedChildren = childrenMap[family];
```